cmd/contour/serve.go

```diff
@@ -405,9 +404,11 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	return g.Run()
 }
 
-func registerEventHandler(informers []cache.SharedIndexInformer, inf cache.SharedIndexInformer, handler cache.ResourceEventHandler) []cache.SharedIndexInformer {
+// registerInformer is a helper to cut boilerplate from registering informers.
+// We need to add the event handler and save the informers HasSynced function for later.
+func registerInformer(inf cache.SharedIndexInformer, handler cache.ResourceEventHandler, sl *SyncList) {
```

Comment:
Hmm, I'm not sure about this function signature, now the append has been factored out I'm even less sure this function makes sense. eg.

It's called `RegisterInformer` but neither of its two lines talk about registering anything. The first add's an event handler ... not an informer. The second adds an informer to something called a SyncList. At least the previous had the, sort of verbose, `informers []cache.SharedIndexInformer` parameter.

What would happen if SyncList.Add() returned the informer it added? then you could have something like list

var informerList SyncList

informerList.Add(coreInformers.Extensions().V1beta1().Ingresses().Informer()).AddEventHandler(handler)

Reply:
That's much better, changed.

Reply:
but it doesn't register an informer, it registers a handler.